libs/sdk/src/common/decorators/tool.decorator.ts

```
@@ -49,86 +58,212 @@ function frontMcpTool<T extends ToolMetadata,
 
-  // ---------- friendly branded errors ----------
-  type __Err<M extends string> = { __type_error__: M } & { never?: never };
+// ---------- friendly branded errors (UPDATED) ----------
 
-  // Must extend ToolContext
-  type __MustExtendCtx<C extends __Ctor> =
-    __R<C> extends ToolContext ? {} :
-      __Err<"Class must extend ToolContext">;
+// Must extend ToolContext (assuming ToolContext is exported by the SDK)
+type __MustExtendCtx<C extends __Ctor> = __R<C> extends ToolContext
+  ? {}
```

ESLint

[error] The `{}` ("empty object") type allows any non-nullish value, including literals like `0` and `""`.
- If that's what you want, disable this lint rule with an inline comment or configure the 'allowObjectTypes' rule option.
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.

(@typescript-eslint/no-empty-object-type)